### ISSUE TITLE
Add grpc-gateway and support for web client

### ISSUE THREAD

**Issue by radu-matei** [radu-matei/kube-toolkit#13]
Right now, there is no officially released support for grpc-web (hopefully soon) - but you have the ability to create a gateway between a web client and the gRPC service, through `.proto` annotations and an HTTP reverse proxy server to your gRPC backend.

Until grpc-web is public (and then there will be a direct implementation of a JavaScript / TypeScript client in the browser), we will use [grpc-gateway](https://github.com/grpc-ecosystem/grpc-gateway)

Essentially, a proto method becomes:

```proto
	rpc GetVersion(google.protobuf.Empty) returns (Version){
		option (google.api.http) = {
            get: "/version"
          };
	}
```

Besides the usual client and server, you also have an automatically generated gateway that you use to start the reverse proxy.

Things to discuss:

- streaming methods - apparently there is support for server streaming, but not over HTTP 1. 
- web client application (framework? to use)
- make the web client optional

Work started in #15

**Comment by radu-matei** [comment]
I'm currently looking at https://github.com/improbable-eng/grpc-web

This allows you to either consume the gRPC server directly from a browser (exactly the app we want), or to have an HTTP proxy that allows any web client and proxies requests to the gRPC server.

**Comment by radu-matei** [comment]
Initially addressed by #15

**Comment by radu-matei** [comment]
Added
